pulp_puppet_plugins/pulp_puppet/plugins/importers/directory.py

```diff
@@ -232,26 +241,31 @@ def _import_modules(self, module_paths):
         :param module_paths: A list of paths to puppet module files.
         :type module_paths: list
         """
-        criteria = UnitAssociationCriteria(type_ids=[constants.TYPE_PUPPET_MODULE],
-                                           unit_fields=Module.UNIT_KEY_NAMES)
-        local_units = self.conduit.get_units(criteria=criteria)
-        local_unit_keys = [unit.unit_key for unit in local_units]
+        existing_module_ids_by_key = {}
+        for module in Module.objects.only(*Module.unit_key_fields).all():
+            existing_module_ids_by_key[module.unit_key_str] = module.id
+
         remote_unit_keys = []
 
         for module_path in module_paths:
             if self.canceled:
                 return
             puppet_manifest = self._extract_metadata(module_path)
-            module = Module.from_json(puppet_manifest)
-            remote_unit_keys.append(module.unit_key())
+            module = Module.from_metadata(puppet_manifest)
+            remote_unit_keys.append(module.unit_key_str)
 
             # Even though we've already basically processed this unit, not doing this makes the
             # progress reporting confusing because it shows Pulp always importing all the modules.
-            if module.unit_key() in local_unit_keys:
+            if module.unit_key_str in existing_module_ids_by_key:
```

Comment:
Here is the issue that I created to track this: https://pulp.plan.io/issues/1300